Modules/Setup

```diff
@@ -166,7 +166,7 @@ _symtable symtablemodule.c
 
 # Modules that should always be present (non UNIX dependent):
 
-#array -DPy_BUILD_CORE_MODULE arraymodule.c	# array objects
+#array -DPy_BUILD_CORE_BUILTIN arraymodule.c	# array objects
```

Comment:
I propose that you revert all changes to ``Modules/Setup``. Python should set macro by default for all modules built by ``Modules/Setup`` and ``Modules/Setup.local``.

Reply:
I you merge GH-25713 first, we can rebase this onto those improvements.

Reply:
See 14504853fd62506f03c8cb1ef7d5e695cac2b31f

Reply:
It looks like ``Makefile.pre.in`` and ``Modules/makesetup`` already set the flag by default. 

``PY_BUILTIN_MODULE_CFLAGS= $(PY_STDMODULE_CFLAGS) -DPy_BUILD_CORE_BUILTIN``

Reply:
Right. `Modules/Setup` could use some clean-up in general as well, then.